File: RA1/JAVA/EjemploPokemon/Combate.java
package EjemploPokemon;

import javax.lang.model.util.ElementScanner6;

import EjemploPokemon.Entrenador;

public class Combate {
    //atributos
    private int puntos_e1, puntos_e2;
    private Entrenador e1, e2;
    private String nombreMedalla;

    //Métodos
    public Combate(Entrenador e1, Entrenador e2, String nombreMedalla){
        this.puntos_e1 = 0;
        this.puntos_e2 = 0;
        this.e1 = e1;
        this.e2 = e2;
        this.nombreMedalla = nombreMedalla;
    }

    public void iniciarCombate(){
        //Recorrer uno a uno los pokemon de cada entrenador y verificar cual de ellos tiene mayor pc
        for(int i = 0; i<3; i++)
            if(e1.getPokemon(i).getPc() >= e2.getPokemon(i).getPc())
                this.puntos_e1++;
            else
                this.puntos_e2++;

        //verificar el ganador con los turnos dados
        if(this.puntos_e1 >= 2)
        {
            System.out.println("El ganador es: " + e1.getNombre());
            e1.ganaMedalla(this.nombreMedalla);
        }
        else
        {
            System.out.println("El ganador es: " + e2.getNombre());
            e2.ganaMedalla(this.nombreMedalla);
        }

    }
}

File: RA1/PYTHON/Pokemon/Combate.py
from Pokemon import Pokemon
from Entrenador import Entrenador

class Combate:
    def __init__(self, e1, e2, nombre_medalla):
        self.puntos_e1 = 0
        self.puntos_e2 = 0

        self.e1 = e1
        self.e2 = e2
        self.nombre_medalla = nombre_medalla

    def iniciar_combate(self):
        for i in range(3):
            if self.e1.get_pokemon(i).get_pc() >= self.e2.get_pokemon(i).get_pc():
                self.puntos_e1 += 1
            else:
                self.puntos_e2 += 1

        if self.puntos_e1 >= 2:
            print("El ganador es: " + self.e1.get_nombre())
            self.e1.gana_medalla(self.nombre_medalla)
        else:
            print("El ganador es: " + self.e2.get_nombre())
            self.e2.gana_medalla(self.nombre_medalla)
